src/background/engineManager.ts

```diff
@@ -544,6 +553,45 @@ export class EngineManager {
     }
     return "ok";
   }
+
+  private lock = new AsyncLock({
+    timeout: 1000,
+  });
+  private lockKey = "write";
+
+  /**
+   * サードパーティ向けの設定ファイルを書き出す
+   */
+  async writeEngineInfoFor3rdParty() {
+    await this.lock.acquire(this.lockKey, async () => {
+      log.info(`Engine information file (runtime-info.json) has been updated.`);
+
+      const engineInfos = this.fetchEngineInfos();
+      const engineInfoList = engineInfos.map((engineInfo) => {
+        return {
+          uuid: engineInfo.uuid,
+          host: engineInfo.host,
+          name: engineInfo.name,
+          path: engineInfo.path,
+          executionEnabled: engineInfo.executionEnabled,
+          executionFilePath: engineInfo.executionFilePath,
+          executionArgs: engineInfo.executionArgs,
+          type: engineInfo.type,
+        };
+      });
+
+      try {
+        await fs.promises.writeFile(
+          this.exportEngineInfoFilename,
+          JSON.stringify(engineInfoList)
```

Comment:
後々他の情報が書き込まれるかもなので、こうかなと！

```suggestion
          JSON.stringify({engineInfos: engineInfoList})
```